src/tribler-core/tribler_core/modules/metadata_store/community/remote_query_community.py

```diff
@@ -130,13 +136,13 @@ def on_send_complete(self, peer, binary_info, binary_data, nonce):
     def on_error(self, peer, exception):
         self.logger.warning(f"EVA transfer error: peer {hexlify(peer.mid)}, exception: {exception}")
 
-    def send_remote_select(self, peer, processing_callback=None, **kwargs):
-
+    def send_remote_select(self, peer, processing_callback=None, force_eva_response=False, **kwargs):
         request = SelectRequest(self.request_cache, hexlify(peer.mid), kwargs, processing_callback)
         self.request_cache.add(request)
 
         self.logger.info(f"Select to {hexlify(peer.mid)} with ({kwargs})")
-        self.ez_send(peer, RemoteSelectPayload(request.number, json.dumps(kwargs).encode('utf8')))
+        payload_class = RemoteSelectPayloadEva if force_eva_response else RemoteSelectPayload
+        self.ez_send(peer, payload_class(request.number, json.dumps(kwargs).encode('utf8')))
```

Comment:
**nit** I know, it is more about personal taste.. but what do you think of this version?

Yes, there are more lines, but it is more obvious.
```suggestion
        args = (request.number, json.dumps(kwargs).encode('utf8'))
        if force_eva_response:
            self.ez_send(peer, RemoteSelectPayloadEva(*args))
        else:
            self.ez_send(peer, RemoteSelectPayload(*args))
```

Reply:
good point